pyrit/prompt_converter/translation_converter.py

```diff
@@ -0,0 +1,85 @@
+import json
+import logging
+import pathlib
+
+from pyrit.interfaces import ChatSupport
+from pyrit.prompt_converter import PromptConverter
+from pyrit.models import PromptTemplate, ChatMessage
+from pyrit.common.path import DATASETS_PATH
+from tenacity import retry, stop_after_attempt, wait_fixed
+
+logger = logging.getLogger(__name__)
+
+
+class TranslationConverter(PromptConverter):
+    def __init__(self, *, converter_target: ChatSupport, languages: list[str], prompt_template: PromptTemplate = None):
+        """
+        Initializes a TranslationConverter object.
+
+        Args:
+            converter_target (ChatSupport): The target chat support for the conversion which will translate
+            language (str): The language for the conversion. E.g. Spanish, French, leetspeak, etc.
+            prompt_template (PromptTemplate, optional): The prompt template for the conversion.
+
+        Raises:
+            ValueError: If the language is not provided.
+        """
+        self.converter_target = converter_target
+
+        # set to default strategy if not provided
+        prompt_template = (
+            prompt_template
+            if prompt_template
+            else PromptTemplate.from_yaml_file(
+                pathlib.Path(DATASETS_PATH) / "prompt_converters" / "translation_converter.yaml"
+            )
+        )
+
+        self._validate_languages(languages)
+
+        self.languages = languages
+        language_str = ", ".join(languages)
+
+        self.system_prompt = str(prompt_template.apply_custom_metaprompt_parameters(languages=language_str))
```

Comment:
nit: `apply_custom_metaprompt_parameters()` will return a string here. We don't need to cast it to a `str` again.
```suggestion
        self.system_prompt = prompt_template.apply_custom_metaprompt_parameters(languages=language_str)
```